compiler/rustc_codegen_llvm/src/llvm/mod.rs

```diff
@@ -331,28 +332,43 @@ pub fn last_error() -> Option<String> {
     }
 }
 
-pub struct OperandBundleDef<'a> {
-    pub raw: &'a mut ffi::OperandBundleDef<'a>,
+/// Owns an [`OperandBundle`], and will dispose of it when dropped.
+pub(crate) struct OperandBundleOwned<'a> {
+    raw: ptr::NonNull<OperandBundle<'a>>,
 }
 
-impl<'a> OperandBundleDef<'a> {
-    pub fn new(name: &str, vals: &[&'a Value]) -> Self {
-        let name = SmallCStr::new(name);
-        let def = unsafe {
-            LLVMRustBuildOperandBundleDef(name.as_ptr(), vals.as_ptr(), vals.len() as c_uint)
+impl<'a> OperandBundleOwned<'a> {
+    pub(crate) fn new(name: &str, vals: &[&'a Value]) -> Self {
```

Comment:
hmm, are you sure this isn't the other way around?
```suggestion
    pub(crate) fn new(name: &str, vals: &'a [&Value]) -> Self {
```

Reply:
The operand bundle retains copies of the individual value pointers, but the slice containing those values is only used temporarily during the constructor call, and is not retained. So `&[&'a Value]` should be correct.

Reply:
Ah, so the slice isn't used, the pointers are. I see.